### PR TITLE
Re add Manufacturer and Supplier Pages

### PR DESCRIPTION
**Pull request by sanctusmob**
Re add Manufacturer and Supplier Pages to Sitemap XML.
For some unknown reason they removed in "Fix all reported bugs #53"
Also increase the number of links and memory usage (my eshop have 40k products)

### CODE REVIEW

File: gsitemap.php [hunk 1]
```diff
@@ -85,7 +87,9 @@ public function install()
             'GSITEMAP_PRIORITY_HOME' => 1.0,
             'GSITEMAP_PRIORITY_PRODUCT' => 0.9,
             'GSITEMAP_PRIORITY_CATEGORY' => 0.8,
-            'GSITEMAP_PRIORITY_CMS' => 0.7,
+            'GSITEMAP_PRIORITY_MANUFACTURER' => 0.7,
+	    'GSITEMAP_PRIORITY_SUPPLIER' => 0.6,
```
Comment (PierreRambaud) [inline gsitemap.php]: Don't mix spaces and tabs :)

File: gsitemap.php [hunk 2]
```diff
@@ -129,6 +133,8 @@ public function uninstall()
             'GSITEMAP_PRIORITY_HOME' => '',
             'GSITEMAP_PRIORITY_PRODUCT' => '',
             'GSITEMAP_PRIORITY_CATEGORY' => '',
+            'GSITEMAP_PRIORITY_MANUFACTURER' => '',
+	    'GSITEMAP_PRIORITY_SUPPLIER' => '',
```
Comment (PierreRambaud) [inline gsitemap.php]: Don't mix spaces and tabs :)

File: gsitemap.php [hunk 3]
```diff
@@ -260,7 +266,7 @@ public function emptySitemap($id_shop = 0)
      */
     public function addLinkToSitemap(&$link_sitemap, $new_link, $lang, &$index, &$i, $id_obj)
     {
-        if ($i <= 25000 && memory_get_usage() < 100000000) {
+        if ($i <= 49000 && memory_get_usage() < 1000000000) {
```
Comment (PierreRambaud) [inline gsitemap.php]: Could you explain the reason to increase these values?

File: gsitemap.php [hunk 4]
```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+	 * @param string $lang            language of link to add
+	 * @param int    $index           index of the current Google Sitemap file
+	 * @param int    $i               count of elements added to sitemap main array
+	 * @param int    $id_manufacturer manufacturer object identifier
+	 *
+	 * @return bool
+	 */
+	protected function getManufacturerLink(&$link_sitemap, $lang, &$index, &$i, $id_manufacturer = 0)
+	{
+		$link = new Link();
```
Comment (PierreRambaud) [inline gsitemap.php]: You should be able to retrieve link from the context? :thinking:

File: gsitemap.php [hunk 5]
```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+		if (method_exists('ShopUrl', 'resetMainDomainCache'))
```
Comment (PierreRambaud) [inline gsitemap.php]: Always use `{` and `}`

File: gsitemap.php [hunk 6]
```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+					Context::getContext()->shop->domain.Context::getContext()->shop->physical_uri
+				), array(
+					'http',
+					Context::getContext()->shop->domain.Context::getContext()->shop->physical_uri.Context::getContext()->shop->virtual_uri
+				), $image_link
+			) : $image_link;
+
+			$file_headers = (Configuration::get('GSITEMAP_CHECK_IMAGE_FILE')) ? @get_headers($image_link) : true;
+			$manifacturer_image = array();
+			if ($file_headers[0] != 'HTTP/1.1 404 Not Found' || $file_headers === true)
```
Comment (PierreRambaud) [inline gsitemap.php]: It's not the right way to retrieve if the response is ok. What if you have a 403, 500, or something similar?

File: gsitemap.php [hunk 7]
```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+				array(
+					'https',
+					Context::getContext()->shop->domain.Context::getContext()->shop->physical_uri
+				), array(
+					'http',
+					Context::getContext()->shop->domain.Context::getContext()->shop->physical_uri.Context::getContext()->shop->virtual_uri
+				), $image_link
+			) : $image_link;
+
+			$file_headers = (Configuration::get('GSITEMAP_CHECK_IMAGE_FILE')) ? @get_headers($image_link) : true;
```
Comment (PierreRambaud) [inline gsitemap.php]: I'm really not in favor of using `@` :/

File: gsitemap.php [hunk 8]
```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+			($this->tableColumnExists(_DB_PREFIX_.'manufacturer_shop') ? ' AND ms.`id_shop` = '.(int)$this->context->shop->id : '').
+			' AND ml.`id_lang` = '.(int)$lang['id_lang'].
+			' ORDER BY m.`id_manufacturer` ASC'
+		);
+		foreach ($manufacturers_id as $manufacturer_id)
+		{
+			$manufacturer = new Manufacturer((int)$manufacturer_id['id_manufacturer'], $lang['id_lang']);
+			$url = $link->getManufacturerLink($manufacturer, $manufacturer->link_rewrite, $lang['id_lang']);
+
+			$image_link = 'http'.(Configuration::get('PS_SSL_ENABLED') && Configuration::get('PS_SSL_ENABLED_EVERYWHERE') ? 's' : '').'://'.Tools::getMediaServer(_THEME_MANU_DIR_)._THEME_MANU_DIR_.((!file_exists(_PS_MANU_IMG_DIR_.'/'.(int)$manufacturer->id.'-medium_default.jpg')) ? $lang['iso_code'].'-default' : (int)$manufacturer->id).'-medium_default.jpg';
```
Comment (PierreRambaud) [inline gsitemap.php]: `Tools::getShopProtocol`

File: gsitemap.php [hunk 9]
```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+			$image_link = (!in_array(rtrim(Context::getContext()->shop->virtual_uri, '/'), explode('/', $image_link))) ? str_replace(
```
Comment (PierreRambaud) [inline gsitemap.php]: This code needs to be refactor, it's the same as above, and doesn't have comments.

File: gsitemap.php (hunk 3 above)
Comment (sanctusmob) [reply gsitemap.php]: my eshop have 40k+ products, if $i is 25000 i get ~half of products.

File: gsitemap.php (hunk 4 above)
Comment (sanctusmob) [reply gsitemap.php]: Done

File: gsitemap.php (hunk 6 above)
Comment (sanctusmob) [reply gsitemap.php]: I am following the same logic with the other get Link functions.

File: gsitemap.php (hunk 7 above)
Comment (sanctusmob) [reply gsitemap.php]: I am following the same logic with the other get Link functions.

File: gsitemap.php (hunk 1 above)
Comment (sanctusmob) [reply gsitemap.php]: I don't see that in my code editor...

File: gsitemap.php (hunk 2 above)
Comment (sanctusmob) [reply gsitemap.php]: I don't see that in my code editor...

File: gsitemap.php [hunk 10]
```diff
@@ -85,7 +87,9 @@ public function install()
             'GSITEMAP_PRIORITY_HOME' => 1.0,
             'GSITEMAP_PRIORITY_PRODUCT' => 0.9,
             'GSITEMAP_PRIORITY_CATEGORY' => 0.8,
-            'GSITEMAP_PRIORITY_CMS' => 0.7,
+            'GSITEMAP_PRIORITY_MANUFACTURER' => 0.7,
+	        'GSITEMAP_PRIORITY_SUPPLIER' => 0.6,
```
Comment (PierreRambaud) [inline gsitemap.php]: Still not the good indentation :)

File: gsitemap.php (hunk 3 above)
Comment (PierreRambaud) [reply gsitemap.php]: Not sure it's the right way to do it.
Maybe people doesn't have the amount of products you have, and the server quality you have :thinking: 
We maybe need to understand why the memory_get_usage has been added.
Also, I think if you have a lot of products, you should use cron task to do it :)

Comment (sanctusmob) [reply gsitemap.php]: I have dedicated server for this eshop only. memory usage for that amount of product and two languages is more than 100mb(and fails to generate XML). 

I have test with 200mb and its ok.

File: gsitemap.php (hunk 2 above)
Comment (sanctusmob) [reply gsitemap.php]: fixed

File: gsitemap.php (hunk 10 above)
Comment (sanctusmob) [reply gsitemap.php]: fixed

File: gsitemap.php (hunk 8 above)
Comment (sanctusmob) [reply gsitemap.php]: obselete

File: gsitemap.php [hunk 11]
```diff
@@ -488,6 +494,131 @@ protected function getCategoryLink(&$link_sitemap, $lang, &$index, &$i, $id_cate
...
+	 * @param int    $i            count of elements added to sitemap main array
+	 * @param int    $id_supplier  supplier object identifier
+	 *
+	 * @return bool
+	 */
+	protected function getSupplierLink(&$link_sitemap, $lang, &$index, &$i, $id_supplier = 0)
+	{
+		$link = new Link();
+		if (method_exists('ShopUrl', 'resetMainDomainCache')) {
+            ShopUrl::resetMainDomainCache();
```
Comment (PierreRambaud) [inline gsitemap.php]: Could you please you use spaces everywhere, indentation is broken.

**Comment by colinegin** [comment]
Hello @sanctusmob , you received some feedbacks about your pull request, are you able to make the required changes ? 

Thanks !

File: gsitemap.php (hunk 3 above)
Comment (mvorisek) [reply gsitemap.php]: keep original value

File: gsitemap.php (hunk 11 above)
Comment (sanctusmob) [reply gsitemap.php]: Done

**Comment by sanctusmob** [comment]
Check again

**Comment by PierreRambaud** [comment]
> Check again

Seems people ignored this PR for a long time, and we are sorry about this :( 
Are you able to rebase your PR based on the dev branch?
We fully understand if you don't want to do it.

**Comment by sanctusmob** [comment]
I will

**Pull request closed without merging**